### PR TITLE
Deprecate using unit denominations in combination with hex numbers.

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -2034,6 +2036,19 @@ void TypeChecker::endVisit(Literal const& _literal)
 				"For more information please see https://solidity.readthedocs.io/en/develop/types.html#address-literals"
 			);
 	}
+	if (_literal.isHexNumber() && _literal.subDenomination() != Literal::SubDenomination::None)
+	{
+		if (v050)
+			m_errorReporter.fatalTypeError(
+				_literal.location(),
+				"Hexadecimal numbers cannot be used with unit denominations."
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: We could suggest to use `0x1234 * 1 day` if they really mean it.

**Review by chriseth — APPROVED** [review]
Fine either way.

File: libsolidity/analysis/TypeChecker.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/analysis/TypeChecker.cpp]: Please include that to avoid user complaints 😉

Comment (ekpyron) [reply libsolidity/analysis/TypeChecker.cpp]: OK, see commit https://github.com/ethereum/solidity/pull/3628/commits/454b470ceed799cb22ceaef6ea2b0bbea7165df9.

File: libsolidity/analysis/TypeChecker.cpp [hunk 2]
```diff
@@ -2034,6 +2036,21 @@ void TypeChecker::endVisit(Literal const& _literal)
 				"For more information please see https://solidity.readthedocs.io/en/develop/types.html#address-literals"
 			);
 	}
+	if (_literal.isHexNumber() && _literal.subDenomination() != Literal::SubDenomination::None)
+	{
+		if (v050)
+			m_errorReporter.fatalTypeError(
+				_literal.location(),
+				"Hexadecimal numbers cannot be used with unit denominations. "
+				"You can use an expression of the form '0x1234 * 1 day' instead."
```
Comment (axic) [inline libsolidity/analysis/TypeChecker.cpp:2045]: Just as a note, usually we've used double quotes for all these, i.e. `form \"0x1234 * 1 day\" instead`

Comment (chriseth) [reply libsolidity/analysis/TypeChecker.cpp:2045]: @ekpyron can you please make this small change?

**Pull request merged**